crypto/keys/keybase_base.go

```diff
@@ -111,29 +145,21 @@ func (kb baseKeybase) DecodeSignature(info Info, msg []byte) (sig []byte, pub tm
 
 // CreateAccount creates an account Info object.
 func (kb baseKeybase) CreateAccount(
-	keyWriter keyWriter, name, mnemonic, bip39Passwd, encryptPasswd string, account, index uint32,
-) (Info, error) {
-
-	hdPath := CreateHDPath(account, index)
-	return kb.Derive(keyWriter, name, mnemonic, bip39Passwd, encryptPasswd, *hdPath)
-}
-
-func (kb baseKeybase) persistDerivedKey(
-	keyWriter keyWriter, seed []byte, passwd, name, fullHdPath string,
+	keyWriter keyWriter, name, mnemonic, bip39Passphrase, encryptPasswd, hdPath string, algo SigningAlgo,
```

Comment:
is the hdPath validated somewhere?

Reply:
It's up to the SigningAlgo's derive function to verify the HD Path

Reply:
BIP32 validity is verified in the secp256k1's derive function:  https://github.com/cosmos/cosmos-sdk/pull/5439/files#diff-317b6cdeb37f9513609b8e3b24e63e32R249